frontend/public/components/cluster-settings/openid-idp-form.tsx

```diff
@@ -0,0 +1,320 @@
+          <div className="help-block" id="mapping-method-description">
+            { /* TODO: Add doc link when available in 4.0 docs. */ }
+            Specifies how new identities are mapped to users when they log in.
+          </div>
+        </div>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="clientID">ClientID</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={clientID}
+            aria-describedby="idp-clientid-help"
+            id="clientID"
+            required />
+        </div>
+        <div className="form-group">
+          <DroppableFileInput
+            onChange={this.clientSecretFileChanged}
+            inputFileData={clientSecretFileContent}
+            id="clientsecret-file"
+            label="Client secret"
+            isRequired
+            hideContents />
+        </div>
+        <div className="co-form-section__separator"></div>
+        <label className="control-label">Claims</label>
+        <p className="co-help-text">The first non-empty claim is used. At least one claim is required.</p>
+        <div className="form-group">
+          <label className="control-label" htmlFor="claimUserName">Preferred username</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={preferredUserName}
+            aria-describedby="idp-claimusername-help"
+            id="preferredUserName" />
+        </div>
+        <div className="form-group">
+          <label className="control-label" htmlFor="claimName">Name</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={claimName}
+            aria-describedby="idp-claimname-help"
+            id="claimName" />
+        </div>
+        <div className="form-group">
+          <label className="control-label" htmlFor="claimEmail">Email</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={claimEmail}
+            aria-describedby="idp-claimemail-help"
+            id="claimEmail" />
+        </div>
+        <div className="co-form-section__separator"></div>
+        <label className="control-label">URLs</label>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="authURL">Authorize URL</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.oidTextValueChanged}
+            value={authorize}
+            aria-describedby="idp-claimusername-help"
+            id="authorize"
+            required />
+        </div>
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="tokenURL">Token URL</label>
```

Comment:
`htmlFor` doesn't match `id` here and below, although these URLs are changing anyway